From 20bba55bcc351cf46430dccf5f3edc5b715fdd5b Mon Sep 17 00:00:00 2001
From: Rocky Hao <rocky.hao@rock-chips.com>
Date: Wed, 27 Jul 2016 16:46:31 +0800
Subject: [PATCH] arm64: dts: rk3399: ajust ipa parameters

Big cores' power consumption and gpu's are greatly reduced, ipa parameters
are ajusted accordingly.

Change-Id: Ibfdae1856c2f1c2cf80a0a2f963ec878053c6ef0
Signed-off-by: Rocky Hao <rocky.hao@rock-chips.com>
---
 .../boot/dts/rockchip/rk3399-evb-rev1.dtsi    | 37 +++++++++++++++++++
 .../boot/dts/rockchip/rk3399-evb-rev2.dtsi    | 37 +++++++++++++++++++
 arch/arm64/boot/dts/rockchip/rk3399.dtsi      | 14 +++----
 3 files changed, 81 insertions(+), 7 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi
index 9a609d77dfa1..7fe93200650d 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev1.dtsi
@@ -56,6 +56,43 @@
 	};
 };
 
+&cpu_l0 {
+	dynamic-power-coefficient = <121>;
+};
+
+&cpu_b0 {
+	dynamic-power-coefficient = <1068>;
+};
+
+&soc_thermal {
+	sustainable-power = <1600>; /* milliwatts */
+
+	cooling-maps {
+		map0 {
+			trip = <&target>;
+			cooling-device =
+				<&cpu_l0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+			contribution = <10240>;
+		};
+		map1 {
+			trip = <&target>;
+			cooling-device =
+				<&cpu_b0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+			contribution = <1024>;
+		};
+		map2 {
+			trip = <&target>;
+			cooling-device =
+				<&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+			contribution = <10240>;
+		};
+	};
+};
+
+&gpu_power_model {
+	dynamic-power = <1780>;
+};
+
 &i2c0 {
 	mp8865: mp8865@68 {
 		compatible = "mps,mp8865";
diff --git a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
index 1d1f41063f81..89043adc5dbe 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399-evb-rev2.dtsi
@@ -55,6 +55,14 @@
 	};
 };
 
+&cpu_l0 {
+	dynamic-power-coefficient = <121>;
+};
+
+&cpu_b0 {
+	dynamic-power-coefficient = <1068>;
+};
+
 &cluster0_opp {
 	opp@408000000 {
 		opp-hz = /bits/ 64 <408000000>;
@@ -194,6 +202,35 @@
 	>;
 };
 
+&soc_thermal {
+	sustainable-power = <1600>; /* milliwatts */
+
+	cooling-maps {
+		map0 {
+			trip = <&target>;
+			cooling-device =
+				<&cpu_l0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+			contribution = <10240>;
+		};
+		map1 {
+			trip = <&target>;
+			cooling-device =
+				<&cpu_b0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+			contribution = <1024>;
+		};
+		map2 {
+			trip = <&target>;
+			cooling-device =
+				<&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
+			contribution = <10240>;
+		};
+	};
+};
+
+&gpu_power_model {
+	dynamic-power = <1780>;
+};
+
 &i2c0 {
 	vdd_cpu_b: syr827@40 {
 		compatible = "silergy,syr827";
diff --git a/arch/arm64/boot/dts/rockchip/rk3399.dtsi b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
index d51384898111..a8cc5bd22bdc 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399.dtsi
+++ b/arch/arm64/boot/dts/rockchip/rk3399.dtsi
@@ -114,7 +114,7 @@
 			reg = <0x0 0x0>;
 			enable-method = "psci";
 			#cooling-cells = <2>; /* min followed by max */
-			dynamic-power-coefficient = <121>;
+			dynamic-power-coefficient = <100>;
 			clocks = <&cru ARMCLKL>;
 			cpu-idle-states = <&cpu_sleep>;
 			operating-points-v2 = <&cluster0_opp>;
@@ -160,7 +160,7 @@
 			reg = <0x0 0x100>;
 			enable-method = "psci";
 			#cooling-cells = <2>; /* min followed by max */
-			dynamic-power-coefficient = <1068>;
+			dynamic-power-coefficient = <436>;
 			clocks = <&cru ARMCLKB>;
 			cpu-idle-states = <&cpu_sleep>;
 			operating-points-v2 = <&cluster1_opp>;
@@ -712,7 +712,7 @@
 		soc_thermal: soc-thermal {
 			polling-delay-passive = <20>; /* milliseconds */
 			polling-delay = <1000>; /* milliseconds */
-			sustainable-power = <1600>; /* milliwatts */
+			sustainable-power = <1000>; /* milliwatts */
 
 			thermal-sensors = <&tsadc 0>;
 
@@ -739,7 +739,7 @@
 					trip = <&target>;
 					cooling-device =
 						<&cpu_l0 THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
-					contribution = <10240>;
+					contribution = <4096>;
 				};
 				map1 {
 					trip = <&target>;
@@ -751,7 +751,7 @@
 					trip = <&target>;
 					cooling-device =
 						<&gpu THERMAL_NO_LIMIT THERMAL_NO_LIMIT>;
-					contribution = <10240>;
+					contribution = <4096>;
 				};
 			};
 		};
@@ -1341,12 +1341,12 @@
 		power-off-delay-ms = <200>;
 		status = "disabled";
 
-		power_model {
+		gpu_power_model: power_model {
 			compatible = "arm,mali-simple-power-model";
 			voltage = <900>;
 			frequency = <500>;
 			static-power = <300>;
-			dynamic-power = <1780>;
+			dynamic-power = <396>;
 			ts = <32000 4700 (-80) 2>;
 			thermal-zone = "gpu-thermal";
 		};
-- 
2.35.3

